python/ray/_raylet.pyx

```diff
@@ -689,20 +689,20 @@ cdef class CoreWorker:
             c_vector[CTaskArg] args_vector
             c_vector[CObjectID] return_ids
 
-        with profiling.profile("submit_task"):
-            prepare_resources(resources, &c_resources)
-            task_options = CTaskOptions(num_return_vals, c_resources)
-            ray_function = CRayFunction(
-                LANGUAGE_PYTHON, string_vector_from_list(function_descriptor))
-            prepare_args(args, &args_vector)
-
-            with nogil:
-                check_status(self.core_worker.get().SubmitActorTask(
-                      c_actor_id,
-                      ray_function,
-                      args_vector, task_options, &return_ids))
-
-            return VectorToObjectIDs(return_ids)
+        # with profiling.profile("submit_task"):
```

Comment:
is removing profiling here ok? Is the profiling part a latency bottleneck?

Reply:
yeah this was just for testing, I'm bringing it back :)

Turns out if we replace the event loop call with a lock, it is fast.